compiler/src/dotty/tools/dotc/transform/InstrumentCoverage.scala

```diff
@@ -273,24 +273,34 @@ class InstrumentCoverage extends MacroTransform with IdentityDenotTransformer:
      * should not be changed to {val $x = f(); T($x)}(1) but to {val $x = f(); val $y = 1; T($x)($y)}
      */
     private def needsLift(tree: Apply)(using Context): Boolean =
-      def isBooleanOperator(fun: Tree) =
-        // We don't want to lift a || getB(), to avoid calling getB if a is true.
-        // Same idea with a && getB(): if a is false, getB shouldn't be called.
-        val sym = fun.symbol
-        sym.exists &&
+      inline def isShortCircuitedOp(sym: Symbol) =
         sym == defn.Boolean_&& || sym == defn.Boolean_||
 
-      def isContextual(fun: Apply): Boolean =
-        val args = fun.args
-        args.nonEmpty && args.head.symbol.isAllOf(Given | Implicit)
+      inline def isCompilerIntrinsic(sym: Symbol) =
```

Comment:
This definition could be moved to the companion object of StringInterpolatorOpt, and used in that phase at https://github.com/lampepfl/dotty/blob/b0671c030bbf9803b5f340b9cc731e48a9b1a5fd/compiler/src/dotty/tools/dotc/transform/localopt/StringInterpolatorOpt.scala#L34 for example.

Reply:
done :)